lightning/rest.go

```diff
@@ -292,15 +289,51 @@ func (h *HTTPAPI) HTTPListPayments(ctx context.Context, req *http.Request, input
 	return &data, nil
 }
 
-// GetHTTPRequest - generic method for doing HTTP requests
-func (h *HTTPAPI) GetHTTPRequest(getData GetDataCall) (*http.Request, *http.Transport, error) {
+// HTTPPeers - invokes peers method.
+func (h *HTTPAPI) HTTPPeers(ctx context.Context, req *http.Request, input *ConnectPeerRequestOverride) error {
+	req = req.WithContext(ctx)
```

Comment:
consider creating common function for Post requests. Most of this one is copy/pasted 6 times now

Reply:
👍 Actually I know why I avoided that before, without generics you can't really make the marshaller work fine, if you pass in an interface it doesn't work as expected. Fixed that now.